docs-src/typescript/chapter-durable-execution/background-replay-intrinsic-non-determinism.md

```diff
@@ -0,0 +1,36 @@
+---
+id: backgroundcheck-replay-intrinsic-non-determinism
+title: Intrinsic non-deterministic logic
+sidebar_label: intrinsic-non-deterministic-logic
+description: This kind of logic prevents the Workflow code from executing to completion because the Workflow can take a different code path than the one expected from the Event History.
+tags:
+- tests
+- replay
+- event history
+---
+
+Referred to as "intrinsic non-determinism" this kind of "bad" Workflow code can prevent the Workflow code from completing because the Workflow can take a different code path than the one expected from the Event History.
+
+In the Temporal TypeScript SDK, certain practices and patterns must be adapted to ensure deterministic and reliable Workflow Execution:
+
+- **Random Number Generation**:
+  - `Math.random()` is overridden by a deterministic version. It uses a pseudo-random number generator seeded with a value specific to the Workflow Execution.
+  - For truly random numbers not tied to the Workflow's deterministic nature, use Activities.
+
+- **Interacting with External Systems or State**:
+  - Directly accessing or mutating external systems or state, such as making API calls, performing file I/O operations, or communicating with other services, should be avoided within the Workflow.
```

Comment:
This is essentially impossible to do in TS, thanks to the Workflow sandbox. We should actually emphasis that fact, as users often struggle to understand why they can't use package X from their Workflow code, the reason being that this package is trying to import a module that does that kind of forbidden operations.